grassroot-webapp/src/main/java/za/org/grassroot/webapp/controller/rest/user/UserController.java

```diff
@@ -161,4 +169,16 @@ public ResponseEntity obtainTokenForApi(HttpServletRequest request) {
         return ResponseEntity.ok(jwtService.createJwt(jwtRequest));
     }
 
+    @RequestMapping(value = "/user/location",method = RequestMethod.POST)
+    public ResponseEntity createUserLocationLog(@RequestParam double lat,
+                                                @RequestParam double lon,
+                                                HttpServletRequest request){
+        log.info("Recieved coodinates, lat = {} , long = {}",lat,lon);
+
+        String userUid = getUserIdFromRequest(request);
+        GeoLocation geoLocation = new GeoLocation(lat,lon);
+
+        userLocationLogRepository.save(new UserLocationLog(Instant.now(),userUid,geoLocation, LocationSource.LOGGED_APPROX));
```

Comment:
I have said before, that controllers should not call repositories directly. Do this through the appropriate broker (up to you to find it, it will be in either async user logger or one of the other brokers).